uswds/uswds-site
# USWDS-Site - Spacing token - Repair converted tokens

## Summary

## Related issue
Closes #1934

## Preview links
- [Tokens - Spacing units](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-spacing-token-05/design-tokens/spacing-units/)
- [Tokens - Flex](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-spacing-token-05/design-tokens/flex/)
- [Tokens - Order](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-spacing-token-05/design-tokens/order/)
- [Tokens - Shadow](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-spacing-token-05/design-tokens/shadow/)

## Problem statement
On the spacing tokens page, the token `5` is incorrectly linked to the `0.5` number equivalent. `0.5` should instead be linked to token `'05'`.  This appears to happen because of confusion with the leading zero and quotes in `'05'` &ndash; something is getting lost in translation between the files that prevents `'05'` from receiving any `converted` value. `5` receives the converted value instead. 

![image](https://user-images.githubusercontent.com/93996430/221238460-4b8b8868-489d-4d72-a97d-6b9d84870584.png)

On the spacing unit page, we need to both:
1. Prevent the `5` token from receiving the `0.5` number equivalent AND 
1. Associate the `'05'` token with the `0.5` number equivalent

Additionally, the  following pages also need to break the association between the `5` token and the `0.5` conversion:
- [Order tokens](https://designsystem.digital.gov/design-tokens/order/)
- [Flex tokens](https://designsystem.digital.gov/design-tokens/flex/)
- [Shadow tokens](https://designsystem.digital.gov/design-tokens/shadow/)

## Solution
The `converted` assignment in `is_number.html` consistently ignored the leading zero in the `'05'` token and would assign the `converted` value to `5` instead. Attempts to explicitly declare `'05'` as a string to force the leading zero were not successful.

As a workaround, this PR creates an explicit `conversion` value for `'05'`  in `spacing.yml`. Declaring an explicit value here removes the confusion that is currently happening between files. 

The spacing unit page appears to be the only instance of `'05'` token that is problematic.

## Screenshots
Final result:

![image](https://user-images.githubusercontent.com/93996430/221237805-48247070-ea44-4c5b-a515-5608dd3706db.png)

## Testing and review
To review:
- Confirm that the `5` token does not have the `0.5` number equivalent on any of the preview pages.
- Confirm that the `'05'` has the `0.5` equivalent on the spacing tokens page.
- Confirm that all pages that use `{{ converted }}` display correct values.
